canonical-web-and-design/maas-ui
# feat(legacy): add host NUMA node id to VM details page, if it exists

## Done

- Added host NUMA node id to VM details page, if it exists.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the machine details page of a VM.
- Check that under `Host` it says `Not NUMA-aligned on ${podName}`.
- The api sends fake machine data for the pod NUMA details, so if you edit the `getPodNumaId` function by replacing `node.system_id` with the system_id of one of the fake machines, it should say `On NUMA node ${ node_id } of ${ podName }`

## Fixes

Fixes canonical-web-and-design/MAAS-squad#1876

## Launchpad issue

Related Launchpad maas issue in the form `lp#number`.

## Backports

In general, please propose fixes against *master* rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in master.
